lib/io/hardware_i2c.cpp

```diff
@@ -69,11 +69,31 @@ std::optional<std::uint8_t> HardwareI2c::readByte(const std::uint8_t device_addr
 core::Result HardwareI2c::writeByteToRegister(const std::uint8_t device_address,
                                               const std::uint8_t register_address,
                                               const std::uint8_t data)
+{
+  const std::uint8_t register_address_array[2] = {0x00, register_address};
```

Comment:
this sneakily wouldn't work, it would write `register_address` to register `0x00`

Reply:
If this wouldn't work for the reason I think you're suggesting (that the eventual use of this array in `write()` will take the first element as the register to write to and the second as the value to write to the array), why would this work for the 16-bit addresses used later which are formed in the same way?

If I have a register address `0x0123` that I want to write the value `0x45` to this register, the code as I've written it will send this to `write()` as a buffer `{0x01, 0x23, 0x45}`. How can `write()` differentiate between:
- Attempting to access register `0x01` to write the value `0x4523`
- Attempting to access register `0x0123` to write the value `0x45`